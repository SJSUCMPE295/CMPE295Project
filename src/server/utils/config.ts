const config = {
    mongoDB:
        'mongodb+srv://wecare:wecare@wecare.l5tdm.mongodb.net/wecare?retryWrites=true&w=majority',
};

//module.exports = config;
export default config;
