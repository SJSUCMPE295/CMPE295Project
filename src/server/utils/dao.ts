import mongoose from 'mongoose';
import doctorModel from 'models/doctor';
import Users from 'models/user';
import { doctorAppointmentModel } from 'models/doctorAppointment';
import userModel from 'models/user';
export const getAllUsers = async (userIds, query = {}) => {
    const userIdsObjects = (userIds || []).map((id) => {
        if (typeof id === 'string') {
            return new mongoose.Types.ObjectId(id);
        }
        return id;
    });
    Object.keys(query).forEach((key) => {
        if (query[key] === undefined) {
            delete query[key];
        }
    });
    const idsQuery = userIdsObjects.length
        ? {
              _id: {
                  $in: userIdsObjects,
              },
          }
        : {};
    const userQuery = {
        ...query,
        ...idsQuery,
    };
    const users = (await userModel.find(userQuery)) || [];
    return users;
};
export const getDoctorsData = async (user) => {
    const userId = user?._id.toString() || user?.userId;
    if (user?.userMetaData?.isDoctor && userId) {
        return doctorModel.findOne({ userId });
    }
};
export const getUserById = async (id) => {
    const userId = (id?.toString && id.toString()) || id;
    const userIdObject = typeof id === 'string' ? new mongoose.Types.ObjectId(id) : id;
    if (userId) {
        const user = await Users.findOne({ _id: userIdObject });
        if (user && user.toJSON()) {
            const doctor = await doctorModel.findOne({ userId });
            const appointments = await doctorAppointmentModel.find(
                doctor ? { doctorId: userId } : { userId }
            );
            return { ...user.toJSON(), doctor, appointments };
        }
    }
};
export const getUserByIdWithAppointments = async (id) => {
    const user = await getUserById(id);
    const users = (await userModel.find()) || [];
    const usersObject = (users?.toJSON ? users.toJSON() : users).reduce(
        (acc, curr) => ({ ...acc, [curr?._id]: curr }),
        {}
    );
    const appointments =
        user?.appointments?.map &&
        user?.appointments?.map((appointment) => {
            const obj = appointment?.toJSON ? appointment.toJSON() : appointment;
            return {
                ...obj,
                id: appointment?._id || obj?._id || obj?.id,
                user: usersObject[appointment?.userId],
                doctor: usersObject[appointment?.doctorId],
            };
        });
    return { ...user, appointments }
};
export const getUserData = async (user) => {
    const userId = user?._id.toString() || user?.userId;
    if (userId) {
        return getUserById(userId);
    }
};
