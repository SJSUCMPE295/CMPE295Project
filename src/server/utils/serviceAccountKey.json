{
  "type": "service_account",
  "project_id": "cmpe295-wecare",
  "private_key_id": "e5b0d57e872cbfd3370d667397ca77950d5a2a05",
  "private_key": "-----BEGIN PRIVATE KEY-----\nMIIEvgIBADANBgkqhkiG9w0BAQEFAASCBKgwggSkAgEAAoIBAQC5I9JrnxwFU24z\nUGgmTtHGzmUTt1cO05NzH+wZx5jwynUw1SyT0iIeBmZpttezpqYCWezWpGierh4i\n8H7vMpJCedEXff3JbUMdqdq9Jq3MG4/IoXff2PuUmCpr8Vug913zz8kmJm8e3TGM\nzZ8Vm215LtwjD0/SH4WMJNSjweByAkeb+CitrlGV+lUnYs686vjxHC6g98at9To8\ny4mlpwtV5iegeLFhcvnkcCLaXWaZcTjoVOUWRH6t5vlVAcbcC+7NBXIFKj+gHIhL\n0g9qUn77lCC2dQcg4Bd4OKLFRipGEkb2pTirhIhO0BvGnHCu5DBss/qFN3tQMmUW\nvVpsh10dAgMBAAECggEAHr3F+LzIqFG8IKUfRg0cZAhtXCK7P+j9uMX6LCWqTYEw\niFU9O6ZOp/GkG7WSh3AKbKWF3/YYsRB0TJTmXs1k38jeYSo4c7PCyzAi8/lyLkv9\nSfEVuIOM8pV5pcbD0iyA2zULT1xeYx+TVK/n1GykdtTDFMc8F8XIJQcQj8hnfkZ+\nFKpNjDHHMCsQDjiXly/wdvTdedpAQ8ukWExWTUVsonj7peFPvYI9E+oASJRN3n/R\nMJD8B2IbfBlfltjYc3mn02ShNs0vDnOxAsN3yMr0whRTa7JxFDtDOtkb3bwvUyh6\nIeUxvuSatYys3sIFofLEcc0Ms2skrRXZtb7E6Lgi+QKBgQDso+AFsD+4x+BK64Nn\nCHPX0RdGcOGu8XBDeebRl0Y05l2EfOqnu09K11PGd1xorFb9EsxndTEybpsJR+MI\nOo9OSJZVNTDMCpJrqjKlOjXsbSUBK5BJv3mMyUCTakhaJv54sphrP55l8UYyIR2T\n2HgNKrgr19E0ZwydiV5bStNO8wKBgQDISVdF47Ygr+/InaCxD7TqGjNh48b/pUdf\nNs8b29pSyq3N1xNjBmeQOtFzzddLfjZska7GjF6+82JvXSRoPThrs82HKL+tXs10\nCjQ4u2jl/8CgOWflu9Ks+v59Mp8Ha7oM+ZXeN2H6yuXWvuG0tFlC6L3IjMAX8wQG\nm+X32dVHrwKBgQDP4kK6HJROGuDyUCVWjTlfPaMp0Z6hVh7eWCeqti8J59n9utNV\n2JoW0EUbU95NYiDlH++Lb4txl3MBWzKaFv+vsNrILfsZNKxegt0NA689bMCWX5PZ\nBamdQWtGVZcMA7eHhrWZegEH40ZcYXUfXYpRXtb+RRDW7TRK7NOEdZ5eewKBgQCz\nnkJsZTjyu5XZXrO71xAyWmj2QyY2Im2NkEPRAiwbz7RrQ5UZW6cJDBbHOjT4eSop\n18TcPwrBSNhgczGxYiX1QPzzV7H+s2Shynzm2/vJOMHxb0okF3tqsnYEpZ9FMnFa\nLYMgFJ8EIW5aVjloq8zZWozJDKFBWSmlG67zxiGl9QKBgGKTxiY4UavCuOCoifxD\n56AEFIPsT83F1gycwbZz7v9bzhHICgpbauLaDWV7fmEr+k2yi3C1pj3u9Fcq1k18\n8WxGkwHy2js60PRasdYICDFUXqtV+rtHW1uXG2/2BuDMldcgV9UGJz8bplNmntAC\nqPnUWmpxlyVj4mKRCVuKZO99\n-----END PRIVATE KEY-----\n",
  "client_email": "firebase-adminsdk-89jb6@cmpe295-wecare.iam.gserviceaccount.com",
  "client_id": "104114169288411082794",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/firebase-adminsdk-89jb6%40cmpe295-wecare.iam.gserviceaccount.com"
}
