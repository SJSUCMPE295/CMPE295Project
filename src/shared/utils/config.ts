//const serverUrl = 'https://ab3b-71-237-235-43.ngrok.io/api/';
const serverUrl = 'http://localhost:8500/api/';

export default serverUrl;
