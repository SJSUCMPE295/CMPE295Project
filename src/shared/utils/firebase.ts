import { initializeApp } from "firebase/app";
import {getAuth} from 'firebase/auth';

const firebaseConfig = {
  apiKey: "AIzaSyBRhY61TgBU4o_iav8XPGUevbIcO74fmgs",
  authDomain: "cmpe295-wecare.firebaseapp.com",
  projectId: "cmpe295-wecare",
  storageBucket: "cmpe295-wecare.appspot.com",
  messagingSenderId: "181804813353",
  appId: "1:181804813353:web:a22ff89a9c9c1397292038",
  measurementId: "G-JKZYL2KZQV"
};

const app = initializeApp(firebaseConfig);
const auth = getAuth(app);
export default auth;