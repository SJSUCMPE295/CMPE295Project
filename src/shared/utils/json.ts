export const objectWithBoolean = (obj = {}) => ({
    ...Object.keys(obj).reduce(
        (acc, curr) => ({
            ...acc,
            [curr]: obj[curr] === 'true' ? true : obj[curr],
        }),
        {}
    ),
});
export const getDate = (daysFromToday = 0) => {
    let dateNow = new Date(); // Creating a new date object with the current date and time
    if (daysFromToday) {
        const today = new Date();
        const future = new Date(today);
        future.setDate(future.getDate() + daysFromToday);
        dateNow = future;
    }
    const year = dateNow.getFullYear(); // Getting current year from the created Date object
    const monthWithOffset = dateNow.getUTCMonth() + 1; // January is 0 by default in JS. Offsetting +1 to fix date for calendar.
    const month = // Setting current Month number from current Date object
        monthWithOffset.toString().length < 2 // Checking if month is < 10 and pre-prending 0 to adjust for date input.
            ? `0${monthWithOffset}`
            : monthWithOffset;
    const date =
        dateNow.getDate().toString().length < 2 // Checking if date is < 10 and pre-prending 0 if not to adjust for date input.
            ? `0${dateNow.getDate()}`
            : dateNow.getDate();
    return `${year}-${month}-${date}`; // combining to format for defaultValue or value attribute of material <TextField>
};
export const formatDate = (dateString, utcTime = false) => {
    const dateNow = new Date(dateString); // Creating a new date object with the current date and time
    const year = dateNow.getFullYear(); // Getting current year from the created Date object
    const monthWithOffset = dateNow.getUTCMonth() + 1; // January is 0 by default in JS. Offsetting +1 to fix date for calendar.
    const month = // Setting current Month number from current Date object
        monthWithOffset.toString().length < 2 // Checking if month is < 10 and pre-prending 0 to adjust for date input.
            ? `0${monthWithOffset}`
            : monthWithOffset;
    const date =
        dateNow.getDate().toString().length < 2 // Checking if date is < 10 and pre-prending 0 if not to adjust for date input.
            ? `0${dateNow.getDate()}`
            : dateNow.getDate();
    const hour = utcTime ? dateNow.getUTCHours() : dateNow.getHours();
    const minutes =
        String(utcTime ? dateNow.getMinutes() : dateNow.getMinutes()).padStart(2, '0') || '00';
    const time = hour > 0 ? `T${hour}:${minutes}` : '';
    return `${year}-${month}-${date}${time}`; // combining to format for defaultValue or value attribute of material <TextField>
};

export const prettyDate = (selectedTime) => {
    const d = new Date(selectedTime);
    const date = d.toLocaleDateString('en-US', { timeZone: 'UTC' });
    const day = d.toLocaleTimeString('en-US', { timeZone: 'UTC' });
    return d ? `${date}  ${day}` : '';
};
