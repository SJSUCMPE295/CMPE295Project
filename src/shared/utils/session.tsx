import {useHistory} from 'react-router-dom';

const session = () => {
    const history = useHistory();
    if(localStorage.getItem('token') != '') {
        history.push('/app/dashboard', { replace: true });
    } else if(localStorage.getItem('token') == undefined) {
        history.push('/login', { replace: true });
    }
}

export default session;