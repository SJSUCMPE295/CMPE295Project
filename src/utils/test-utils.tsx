import { Provider } from 'react-redux';
import React from 'react';
import configureStore from '../shared/store';

export const withRedux = (Component) => {
    const wrapped: React.FunctionComponent<any> = ({ initialState = {}, store, ...props }) => {
        return (
            <Provider store={store || configureStore({ initialState })}>
                <Component {...props} />
            </Provider>
        );
    };
    wrapped.displayName = Component.displayName || Component.name || 'Component';
    return wrapped;
};
