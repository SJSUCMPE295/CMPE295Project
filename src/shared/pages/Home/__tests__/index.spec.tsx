import React from 'react';
import { cleanup, render, fireEvent } from '@testing-library/react';
import Page from '..';
import { withRedux } from '../../../../utils/test-utils';
import { defaultData } from './fixtures';
const PageWithRedux = withRedux(Page);
describe('<Page {...defaultProps} />', () => {
    afterEach(cleanup);
    it('should render element', () => {
        const { container } = render(<PageWithRedux data={defaultData} />);
        expect(container).toMatchSnapshot();
    });
    it('should render text content', () => {
        const { getByText } = render(<PageWithRedux data={defaultData} />);
        expect(getByText('English')).toBeInTheDocument();
    });
});
