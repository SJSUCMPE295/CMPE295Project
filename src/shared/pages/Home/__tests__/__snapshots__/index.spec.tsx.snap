// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page {...defaultProps} /> should render element 1`] = `
<div>
  <h2>
    features
  </h2>
  <ul>
    <li>
      React 16.x (latest)
    </li>
    <li>
      Webpack 4
    </li>
    <li>
      ESLint 6
    </li>
    <li>
      Babel 7
    </li>
    <li>
      TypeScript (using Babel 7)
    </li>
    <li>
      Jest
    </li>
    <li>
      React Testing Library
    </li>
    <li>
      React Fast Refresh
    </li>
    <li>
      React Router 5
    </li>
    <li>
      Redux (+ Thunk)
    </li>
    <li>
      Immer
    </li>
    <li>
      Reselect
    </li>
    <li>
      React Helmet
    </li>
    <li>
      Express Webserver + Server Side Rendering
    </li>
    <li>
      CSS Modules
    </li>
    <li>
      PostCSS
    </li>
    <li>
      Prettier (incl. precommit-hook via lint-staged + husky)
    </li>
  </ul>
  <p>
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
      value="en_US"
    >
      English
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </p>
</div>
`;
